opencast-backend/annotation-impl/src/main/java/org/opencast/annotation/impl/persistence/ExtendedAnnotationServiceJpaImpl.java

```diff
@@ -1007,47 +1062,20 @@ public ExtendedAnnotationException apply(Exception e) {
    *          value of the ":id" parameter in the named query.
    */
   private <A, B> Option<A> findById(final Function<B, A> toA, final String queryName, final Object id) {
-    return tx(named.<B> findSingle(queryName, id(id))).map(toA);
-  }
-
-  /**
-   * Do not nest inside a tx!
-   *
-   * @param id
-   *          value of the ":id" parameter in the named query.
-   */
-  private <A> Option<A> findById(final String queryName, final Object id) {
-    return tx(named.findSingle(queryName, id(id)));
-  }
-
-  /**
-   * Do not nest inside a tx!
-   *
-   * @param id
-   *          value of the ":id" parameter in the named query.
-   */
-  private <A, B> List<A> findAllById(final Function<B, A> toA, final Option<Integer> offset,
-          final Option<Integer> limit, final String queryName, final Object id) {
-    return tx(named.<B> findAllM(queryName, offset, limit, id(id))).map(toA).value();
-  }
-
-  @SafeVarargs
-  private final <A, B> List<A> findAllWithParams(final Function<B, A> toA, final Option<Integer> offset,
-          final Option<Integer> limit, final String queryName, final Tuple<String, ?>... params) {
-    return tx(named.<B> findAllM(queryName, offset, limit, params)).map(toA).value();
-  }
-
-  private Option<VideoDto> getVideoDto(final long id) {
-    return findById("Video.findById", id);
-  }
-
-  private Option<TrackDto> getTrackDto(final long id) {
-    return findById("Track.findById", id);
+    Optional<B> result = (Optional<B>) tx(em -> {
+      return namedQuery.findOpt(queryName, Pair.of("id", id)).apply(em);
+    });
+    if (result.isPresent()) {
+      A appliedResult = toA.apply(result.get());
+      return some(appliedResult);
+    } else {
+      return none();
+    }
```

Comment:
Sorry, I should have been more clear: I would have kept (and changed) the overload **without** the function.

Plus I would still love to see you use the typesafe version of `findOpt` here instead of a blind cast.

Oh, and can we move this function closer to `findAllWith...`? Maybe together with `id`? Which you could also use here instead of `Pair.of("id", id)`, btw. 😉

Reply:
> Sorry, I should have been more clear: I would have kept (and changed) the overload without the function.

Gonna respectfully disagree on that. I played around with it a bit, and in my opinion it makes the code more convoluted and less type safe.

Reply:
Sad but fair. 🤷‍♀️